mooncake-transfer-engine/src/transfer_metadata_plugin.cpp

```diff
@@ -264,19 +264,28 @@ struct HTTPStoragePlugin : public MetadataStoragePlugin {
 #ifdef USE_ETCD
 struct EtcdStoragePlugin : public MetadataStoragePlugin {
     EtcdStoragePlugin(const std::string &metadata_uri)
-        : client_(metadata_uri), metadata_uri_(metadata_uri) {}
+        : client_(0), metadata_uri_(metadata_uri) {
+        client_ = NewEtcdClient((char *) metadata_uri_.c_str());
+    }
 
-    virtual ~EtcdStoragePlugin() {}
+    virtual ~EtcdStoragePlugin() { 
+        if (client_) {
+            EtcdCloseWrapper(client_);
+            client_ = 0;
+        }
+    }
 
     virtual bool get(const std::string &key, Json::Value &value) {
         Json::Reader reader;
-        auto resp = client_.get(key);
-        if (!resp.is_ok()) {
+        char *json_data = nullptr;
+        auto ret = EtcdGetWrapper(client_, (char *) key.c_str(), &json_data);
+        if (ret || !json_data) {
             LOG(ERROR) << "EtcdStoragePlugin: unable to get " << key << " from "
-                       << metadata_uri_ << ": " << resp.error_message();
+                       << metadata_uri_;
             return false;
         }
-        auto json_file = resp.value().as_string();
+        auto json_file = std::string(json_data);
+        free(json_data);
```

Comment:
better add a comment for this line, for better understanding, i.e.
It's allocated in Golang side through C.CString, so we need to free it here.